### PR TITLE
SYNPY-413: update a view schema

### PR DESCRIPTION
**Pull request by kdaily**

### CODE REVIEW

File: synapseclient/__main__.py [hunk 1]
```diff
@@ -79,7 +79,8 @@
 import warnings
 from .exceptions import *
 import getpass
-
+import csv
```
Comment (kdaily) [inline synapseclient/__main__.py]: do we need to use `backports.csv` like the testing modules do?

File: synapseclient/table.py [hunk 2]
```diff
@@ -538,9 +538,17 @@ def _before_synapse_store(self, syn):
                 self.properties.columnIds.append(column.id)
             self.__dict__['columns_to_store'] = None
 
+class ViewSchema(Schema):
+    _synapse_entity_type = 'org.sagebionetworks.repo.model.table.EntityView'
+    _viewType = 'file'
+
```
Comment (larssono) [inline synapseclient/table.py]: viewType probaby doesn't belong.

File: synapseclient/table.py [hunk 3]
```diff
@@ -538,9 +538,17 @@ def _before_synapse_store(self, syn):
...
+    def _addScope(self):
+        raise NotImplementedError
+
```
Comment (larssono) [inline synapseclient/table.py]: Not this PR

File: synapseclient/table.py [hunk 4]
```diff
@@ -775,7 +783,7 @@ class TableAbstractBaseClass(object):
     Abstract base class for Tables based on different data containers.
     """
     def __init__(self, schema, headers=None, etag=None):
-        if isinstance(schema, Schema):
+        if isinstance(schema, (Schema, ViewSchema)):
```
Comment (larssono) [inline synapseclient/table.py]: Is this necessary as ViewSchema is subclass of Schema?

File: synapseclient/table.py [hunk 5]
```diff
@@ -1047,7 +1055,7 @@ def from_data_frame(cls, schema, df, filepath=None, etag=None, quoteCharacter='"
             headers = [SelectColumn.from_column(col) for col in cols]
 
         ## if the schema has no columns, use the inferred columns
-        if isinstance(schema, Schema) and not schema.has_columns():
+        if isinstance(schema, (Schema, ViewSchema)) and not schema.has_columns():
```
Comment (larssono) [inline synapseclient/table.py]: dito.

File: synapseclient/table.py [hunk 6]
```diff
@@ -1174,12 +1182,12 @@ def __init__(self, schema, filepath, etag=None, quoteCharacter='"', escapeCharac
         self.setColumnHeaders(headers)
 
     def _synapse_store(self, syn):
-        if isinstance(self.schema, Schema) and self.schema.get('id', None) is None:
+        if isinstance(self.schema, (Schema, ViewSchema)) and self.schema.get('id', None) is None:
```
Comment (larssono) [inline synapseclient/table.py]: dito.

File: tests/integration/test_tables.py [hunk 7]
```diff
@@ -44,6 +44,72 @@ def setup(module):
...
+
+    # Add new columns for the annotations on this file and get their IDs
+    my_added_cols = [syn.store(synapseclient.Column(name=k, columnType="STRING")) for k in file_annotations.keys()]
+    my_added_cols_ids = [c['id'] for c in my_added_cols]
+
+    # Required IDs for Synapse properties
+    # Should be defined somewhere in synapseclient.table
+    minimal_view_schema_column_ids = ['2510', '23543', '23544', '23545', '31783', '26823', '31784', '23550',
+                                      '30514', '31785', '31782', '31786']
+
```
Comment (larssono) [inline tests/integration/test_tables.py]: Ask Jay how he finds these instead of hard coding.

File: tests/integration/test_tables.py [hunk 8]
```diff
@@ -44,6 +44,72 @@ def setup(module):
...
+    scopeIds = [folder['id'].lstrip('syn')]
+
```
Comment (larssono) [inline tests/integration/test_tables.py]: file jira to make backend compatible with every other place ids are used.

File: tests/integration/test_tables.py [hunk 9]
```diff
@@ -44,6 +44,72 @@ def setup(module):
...
+        assert view_dict[0][k] == v
+
+    # Make a change to the view and store
+    view_dict[0]['fileFormat'] = 'PNG'
+    tmp_file = tempfile.NamedTemporaryFile(suffix=".csv")
+    dw = csv.DictWriter(tmp_file, fieldnames=view_dict[0].keys())
+    dw.writeheader()
+    dw.writerows(view_dict)
+    tmp_file.flush()
+
```
Comment (larssono) [inline tests/integration/test_tables.py]: Should have convenience fucntions.  Then this should move to that test.

File: tests/integration/test_tables.py (hunk 8 above)
Comment (kdaily) [inline tests/integration/test_tables.py]: https://sagebionetworks.jira.com/browse/PLFM-4344

**Pull request merged**